### PR TITLE
Get information about patch proposals from API

### LINKED ISSUE #195: Migrate Radicle -> Patch Proposals to heartwood

### PR DESCRIPTION
**Pull request by Stelios123**
- Call the API to get the patch proposals
- Listing the patch proposals
- Allow the user to filter based on author / project

closes #195

### CODE REVIEW

**Comment by Stelios123** [comment]
Result : 


https://user-images.githubusercontent.com/31673488/231698198-c3b566d1-8def-449f-9628-b76310abd0eb.mp4

File: src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java [hunk 1]
```diff
@@ -1,45 +1,125 @@
...
+    public List<String> tags;
+    public State state;
     public List<Revision> revisions;
 
-    public RadPatch(GitRepository repo, String patchId, String author, String title, String description, State state, String target,
-                    List<Revision> revisions) {
-        this.repo = repo;
-        this.patchId = patchId;
-        this.author = author;
+    public record Author(String id) {
```
Comment (JChrist) [inline src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java:21]: please move all nested records/enums to the bottom of the class, so they're out of the way when looking at the class fields/methods

File: src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java [hunk 2]
```diff
@@ -1,45 +1,125 @@
...
     }
 
-    public record Revision(String patchId, String description, String base, String commitHash, Instant created, List<String> refs,
-                           List<Comment> comments, List<Merge> merges) {
+    public record Revision(String id, String description, String base, String oid, List<String> refs,
+                           List<Merge> merges, Instant timestamp,
+                           List<Discussion> discussions, List<String> reviews) {
+
+        @Override
+        public String toString() {
```
Comment (JChrist) [inline src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/models/RadPatch.java:67]: this override does not really offer much, it's only a simulation of json (that can easily fail, e.g. in cases of null).
It's better to use jackson straight away for json serialization, so please remove all overriden `toString` from records.

File: src/test/java/network/radicle/jetbrains/radiclejetbrainsplugin/patches/RadicleToolWindowTest.java [hunk 3]
```diff
@@ -28,26 +31,22 @@ public class RadicleToolWindowTest extends AbstractIT {
...
-                "\"refs\":[],\"merges\":[],\"discussions\":[],\"timestamp\":1680864771,\"reviews\":[]}," +
-                "{\"id\":\"14e35ef75f6c7b73f965004f6f985934ba13bcd9\",\"description\":\"Add README, just for the fun\",\"base\":" +
-                "\"9de952f92e27b67b43d89230e2d68caa0794b5a9\",\"oid\":\"84fb6549088427b958903eafa52b6559b20756de\"," +
-                "\"refs\":[\"refs/heads/new_branch\"],\"merges\":[],\"discussions\":[{\"id\":\"c37274314cbe28efdf7c33abcaa01b3473797985\",\"author\":" +
-                "{\"id\":\"did:key:z6MkwM5cMjgsvTcUyEtUAfo1bWzGS9o41zXRcy7eqaZubwzS\"}," +
-                "\"body\":\"TestMsg!\",\"reactions\":[],\"timestamp\":1680867331,\"replyTo\":null},{\"id\":\"538fe004d02e84e554c3b49b70567bdd59cd48c5\"," +
-                "\"author\":{\"id\":\"did:key:z6MkwM5cMjgsvTcUyEtUAfo1bWzGS9o41zXRcy7eqaZubwzS\"}," +
-                "\"body\":\"This is my Reply\",\"reactions\":[],\"timestamp\":1680867502,\"replyTo\":\"c37274314cbe28efdf7c33abcaa01b3473797985\"}]," +
-                "\"timestamp\":1680865493,\"reviews\":[]}]}]");
+        StringEntity se = new StringEntity(getTestPatches().toString());
```
Comment (JChrist) [inline src/test/java/network/radicle/jetbrains/radiclejetbrainsplugin/patches/RadicleToolWindowTest.java:48]: similarly, here, it's better to use jackson to serialize the value, something like:
`new ObjectMapper().findAndRegisterModules().writeValueAsString(getTestPatches());`

**Comment by gsaslis** [comment]
@Stelios123 thanks so much for the screen recording!

One small favour: from now on, can we please add those on the **issue**, instead of the PR ? ;)

**Pull request merged**